infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/RestApi.java

```diff
@@ -65,6 +65,18 @@ protected SafeFuture<?> doStart() {
                 "TCP Port %d is already in use. "
                     + "You may need to stop another process or change the HTTP port for this process.",
                 app.port()));
+      } else if (e instanceof IllegalStateException
+          || Throwables.getRootCause(e) instanceof IllegalStateException) {
+        // IllegalStateException is a sign that something needed has failed to be initialised.
+        // throwing it here will terminate the process effectively.
+        LOG.error("Failed to start Rest API", e);
+        throw e;
+      } else if (app.jettyServer() == null || !app.jettyServer().started) {
```

Comment:
we could move the `app.jettyServer() == null` check right after `app.start()` so we can remove all intelliJ's complains

Reply:
i did try this but its still messy in intellij :(